libbeat/monitoring/report/report.go

```diff
@@ -0,0 +1,127 @@
+package report
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/publisher"
+)
+
+type config struct {
+	// allow for maximum one reporter being configured
+	Reporter common.ConfigNamespace `config:",inline"`
+}
+
+type Reporter interface {
```

Comment:
Seems like we start to have this `Runner` interface in lots of different places. We should probably start to use a shared interface.